display-servers/xlib-display-server/src/event_translate.rs

```diff
@@ -216,3 +233,68 @@ fn from_button_release(x_event: XEvent) -> DisplayEvent {
     xw.set_mode(Mode::Normal);
     DisplayEvent::ChangeToNormalMode
 }
+
+/// # Panics
+/// If it cannot open xrandr.
+/// This should normally only be called if `x_event.0.xrandr_event_base` is `Some`,
+/// indicating a succesful connection to xrandr was made previously.
+fn from_xrandr_crtc_change(x_event: XEvent) -> Option<DisplayEvent> {
+    use leftwm_core::models::Screen;
+    use std::slice;
+
+    let xrandr =
+        xrandr::Xrandr::open().expect("Function ony called if `xrandr_event_base` is some");
+    let event = xrandr::XRRCrtcChangeNotifyEvent::from(x_event.1);
+
+    unsafe {
+        let screen_resources = (xrandr.XRRGetScreenResources)(x_event.0.display, x_event.0.root);
+        let crtc_info = (xrandr.XRRGetCrtcInfo)(x_event.0.display, screen_resources, event.crtc);
+        let outputs = slice::from_raw_parts((*crtc_info).outputs, (*crtc_info).noutput as usize);
+
+        if outputs.len() > 1 {
+            tracing::error!(
+                r#"Leftwm does not support more than one output per crtc (if that is even possible to have). 
+                LeftWM will only apply changes to the first output. 
+                If you are seing this error, please create an issue on our GitHub page and it will be resolved."#
```

Comment:
@VuiMuich @hertg what do you think about that?

I have tried some things, but I haven't goten xrandr yet to put both of my monitors (outputs) on one crtc. It is possible to return a `Vec<Screen>` here, but I don't think this kind of exotic configuration even exists (if possible). I've added this error to indicate an issue. This is not fatal at all, you will just have to SoftReload like before to reset the configuration.

Reply:
I have to admit that I'm a bit too out of the loop to actually understand what you mean by outputs and crtc. But I like your thinking ahead and very verbose log message. I fully approve of, and appreciate your approach.

Reply:
![image](https://user-images.githubusercontent.com/76821666/234400403-f447cb62-79df-4c98-8462-a933aaa0502f.png)
...I am still trying to understand that myself. I haven't found no real documentation on the internal XRandR API whatsoever. I just try to get together what I can.

I think there are the **output**, the **crtc** and the **screen**. The screen is everything covered by the `DISPLAY` env variable, whilst the output corresponds to a hardware monitor and the crtc is basically the virtual representation of the output.